coffee_maker_navigator_2/lib/features/add_water/domain/entities/add_water_state.dart

```diff
@@ -0,0 +1,52 @@
+import 'package:coffee_maker_navigator_2/features/add_water/domain/entities/water_source.dart';
+
+class AddWaterState {
```

Comment:
This new approach aims to avoid creating separate State classes. We should avoid combining multiple fields in one class. For example, the [footer widget](https://github.com/woltapp/wolt_modal_sheet/pull/335/files#diff-34f0c9b96d50aea1eacbcc1faf9efbb3d29dec325400cd4b121a946ab058ab60R42) only depends on `errorMessage` and  `isReadyToAddWater` and it should not be re-built when waterSource for example changes.